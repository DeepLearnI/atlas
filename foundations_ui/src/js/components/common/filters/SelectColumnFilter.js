import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CommonActions from '../../../actions/CommonActions';
import CheckboxFilter from './CheckboxFilter';

const isStatus = false;

class SelectColumnFilter extends Component {
  constructor(props) {
    super(props);
    this.changeLocalParams = this.changeLocalParams.bind(this);
    this.onApply = this.onApply.bind(this);
    this.onCancel = this.onCancel.bind(this);
    this.onClearFilters = this.onClearFilters.bind(this);
    this.updateSearchText = this.updateSearchText.bind(this);
    this.submitSearchText = this.submitSearchText.bind(this);
    this.unsetClearFilters = this.unsetClearFilters.bind(this);
    this.onHideAll = this.onHideAll.bind(this);
    this.unsetHideAll = this.unsetHideAll.bind(this);
    this.state = {
      columns: this.props.columns,
      changeHiddenParams: this.props.changeHiddenParams,
      changedParams: this.props.hiddenInputParams,
      toggleShowingFilter: this.props.toggleShowingFilter,
      updateSearchText: this.props.updateSearchText,
      searchText: '',
      showAllFilters: false,
      hideAllFilters: false,
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ columns: nextProps.columns });
  }

  onHideAll() {
    const { columns } = this.props;
    const filteredColumns = columns.map(filter => {
      return filter.name;
    });
    this.setState({ changedParams: filteredColumns, hideAllFilters: true, showAllFilters: false });
  }

  unsetHideAll() {
    this.setState({ hideAllFilters: false });
  }

  onApply() {
    const {
      changeHiddenParams, changedParams, toggleShowingFilter, updateSearchText,
    } = this.state;
    changeHiddenParams(changedParams);
    toggleShowingFilter();
    updateSearchText('');
  }

  onCancel() {
    const { toggleShowingFilter, updateSearchText } = this.state;
    this.setState({ changedParams: [] });
    toggleShowingFilter();
    updateSearchText('');
  }

  onClearFilters() {
    const { updateSearchText } = this.state;
    const emptyArray = [];
    this.setState({ changedParams: emptyArray, showAllFilters: true, hideAllFilters: false });
    updateSearchText('');
    this.input.value = '';
  }

  unsetClearFilters() {
    this.setState({ showAllFilters: false });
  }

  changeLocalParams(colName) {
    const { changedParams } = this.state;
    const copyArray = CommonActions.getChangedCheckboxes(changedParams, colName);
    this.setState({ changedParams: copyArray });
  }

  updateSearchText(e) {
    this.setState({ searchText: e.target.value });
  }

  submitSearchText() {
    const { searchText, updateSearchText } = this.state;
    updateSearchText(searchText);
  }

  render() {
    const { columns, showAllFilters, hideAllFilters } = this.state;
    const checkboxes = CommonActions.getCheckboxes(
      columns, this.changeLocalParams, showAllFilters, this.unsetClearFilters, hideAllFilters, isStatus,
      this.unsetHideAll,
    );

    const input = <input ref={inputRef => { this.input = inputRef; }} type="text" onChange={this.updateSearchText} />;

    return (
      <CheckboxFilter
        checkboxes={checkboxes}
        onCancel={this.onCancel}
        onApply={this.onApply}
        submitSearchText={this.submitSearchText}
        onClearFilters={this.onClearFilters}
        input={input}
        onHideAll={this.onHideAll}
      />
    );
  }
}

SelectColumnFilter.propTypes = {
  columns: PropTypes.array,
  changeHiddenParams: PropTypes.func,
  changedParams: PropTypes.array,
  toggleShowingFilter: PropTypes.func,
  hiddenInputParams: PropTypes.array,
  updateSearchText: PropTypes.func,
  showAllFilters: PropTypes.bool,
  hideAllFilters: PropTypes.bool,
};

SelectColumnFilter.defaultProps = {
  columns: [],
  changeHiddenParams: () => {},
  changedParams: [],
  toggleShowingFilter: () => {},
  hiddenInputParams: [],
  updateSearchText: () => {},
  showAllFilters: false,
  hideAllFilters: false,
};

export default SelectColumnFilter;
