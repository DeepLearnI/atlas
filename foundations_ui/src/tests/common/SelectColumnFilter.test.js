import React from 'react';
import ReactDOM from 'react-dom';
import SelectColumnFilter from '../../js/components/common/filters/SelectColumnFilter';
import { shallow, mount } from 'enzyme';
import configureTests from '../setupTests';

configureTests();

it('Shallow Renders SelectColumnFilter', () => {
  const wrapper = shallow(<SelectColumnFilter/>);
  expect(wrapper).toMatchSnapshot();
});

it('Change input selectColumnFilter', () => {
  const wrapper = mount(<SelectColumnFilter/>);
  const input = wrapper.find('input').getDOMNode();
  const button = wrapper.find('button').at(2);
  input.value = 'Test';
  expect(input.value).toBe('Test');
  wrapper.instance().onClearFilters();
  const updatedInput = wrapper.find('input').getDOMNode();
  expect(updatedInput.value).toBe('');
});