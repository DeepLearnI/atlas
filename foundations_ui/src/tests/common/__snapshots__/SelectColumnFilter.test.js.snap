// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders SelectColumnFilter 1`] = `
<CheckboxFilter
  addedClass=""
  applyClass="b--mat b--affirmative text-upper"
  checkboxes={Array []}
  input={
    <input
      onChange={[Function]}
      type="text"
    />
  }
  onApply={[Function]}
  onCancel={[Function]}
  onClearFilters={[Function]}
  onHideAll={[Function]}
  submitSearchText={[Function]}
/>
`;
